test/hooks.test.js

```diff
@@ -714,7 +714,7 @@ test('onRoute hook should able to change the route url', t => {
   fastify.register((instance, opts, done) => {
     instance.addHook('onRoute', (route) => {
       t.equal(route.url, '/föö')
-      route.url = encodeURI(route.url)
+      route.url = '/föö2'
```

Comment:
Why this change?

Reply:
Since we apply `decodeURI` to each url right after we get a request it becomes unnecessary and useless to `encodeURI` for routes. The test name was `onRoute hook should able to change the route url` so I kept the test to be doing what's expected. The `onRoute` hook does change the route url and the test makes sure that this works.

But I think I am going to close this PR because it looks like it has to be fixed in `find-my-way` lib. I also noticed that `fast-decode-uri-component` works faster only when there is at least 25% of malformed URIs. When there is no malformed URIs the standard `decodeURIComponent` works up to 2 times faster (especially if you add a precondition ```if (uri.indexOf('%') < 0) return uri``` before calling the `decodeURIComponent`)